### ISSUE TITLE
QuickSight group membership

### ISSUE THREAD

**Issue by with-joy** [hashicorp/terraform-provider-aws#16609]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Potentially a new QuickSight resource to create group membership for a user.
### New or Affected Resource(s)

* aws_quicksight_group_membership

### Potential Terraform Configuration

```hcl
resource "aws_quicksight_group_membership" "example" {
  group_name = "tf-example"
  user_name  = "an-author"
}
```

### References

https://docs.aws.amazon.com/cli/latest/reference/quicksight/create-group-membership.html

**Comment by with-joy** [comment]
@anGie44 , is it ok if I try to open a PR for this?

**Comment by anGie44** [comment]
Hi @with-joy , we certainly welcome a PR for this feature! Please note though this service is not at the forefront of the Provider roadmap (reference: https://github.com/hashicorp/terraform-provider-aws/blob/master/ROADMAP.md), and this feature request is still relatively new to have acquired more community interest, so it may be some time before your PR reaches our team's review cycle and this may entail a rebase here and there as the code base changes. With that said, if you are still inclined, please feel free to give it a go and we look forward to reviewing it in the future :)

**Comment by anGie44** [comment]
Hi @with-joy 👋  , noting here that the group membership resource been merged (https://github.com/hashicorp/terraform-provider-aws/pull/20687) and is available since `v3.58.0` of the AWS Provider.
